src/Avalonia.Controls/Primitives/TextSelectionCanvas.cs

```diff
@@ -279,21 +294,14 @@ internal void SetPresenter(TextPresenter? textPresenter)
                     _textBox.SizeChanged += TextBox_SizeChanged;
                 }
             }
-            else
-            {
-                if (_textBox != null)
-                {
-                    _textBox.RemoveHandler(TextBox.TextChangingEvent, TextChanged);
-                    _textBox.RemoveHandler(KeyDownEvent, TextBoxKeyDown);
-                    _textBox.RemoveHandler(PointerReleasedEvent, TextBoxPointerReleased);
-                    _textBox.RemoveHandler(Gestures.HoldingEvent, TextBoxHolding);
-
-                    _textBox.PropertyChanged -= TextBoxPropertyChanged;
-                    _textBox.EffectiveViewportChanged -= TextBoxEffectiveViewportChanged;
-                    _textBox.SizeChanged -= TextBox_SizeChanged;
-                }
+        }
 
-                _textBox = null;
+        private void Presenter_CaretBoundsChanged(object? sender, EventArgs e)
```

Comment:
As far as I can tell, this new method isn't referenced anywhere?

Reply:
Removed.